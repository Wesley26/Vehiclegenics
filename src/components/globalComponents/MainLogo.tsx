import { FC } from 'react';
import { Text, View } from 'react-native';
import { globalStyles } from '../../../globalStyles';
import { useTailwind } from 'tailwind-rn';

import DetermineDevice from '../../hooks/anyHooks/DetermineDevice';

/**
 * @returns MainLogo component
 */

const MainLogo:FC = () => {

    const deviceType: number = DetermineDevice();
    const tailwind = useTailwind();
    const logoText: string = `Vehiclegenics`;

    console.log(`The deviceType inside MainLogo.tsx is: ${(deviceType)}`);
    //todo: create the tailwind styles for both tablet and phone

    return (

        <View style={[
            globalStyles.bgMedium,
            tailwind('mt-55pix p-15pix h-32 border-slate-700 border-3pix justify-center items-center rounded-md')
        ]}>
            <View style={[
                globalStyles.textDark, 
                tailwind('justify-center items-center')]}
            >
                <Text style={[
                    globalStyles.workSansBold700,
                    globalStyles.primaryShadow, 
                    tailwind('mt-2pix px-10pix justify-center items-center text-42pix text-center')]}
                >
                    {logoText}
                </Text>
            </View>
        </View>

    );
};

export default MainLogo;