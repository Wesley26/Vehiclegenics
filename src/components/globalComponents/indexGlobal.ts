/**
 * Export all global components from this
 * index file
 */

export { default as MainLogo } from './MainLogo';