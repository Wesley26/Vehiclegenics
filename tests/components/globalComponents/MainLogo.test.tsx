import React from "react";
import renderer from "react-test-renderer";
import { MainLogo } from '../../../src/components/globalComponents/indexGlobal';

//required so tailwindcss + tailwind-rn play nice with test
jest.useFakeTimers();

describe (`describing MainLogo.tsx component`, () => {

    it(`render MainLogo.tsx test component`, () => {

        const view = renderer.create(<MainLogo />);
        expect(view).toMatchSnapshot();

    });

});