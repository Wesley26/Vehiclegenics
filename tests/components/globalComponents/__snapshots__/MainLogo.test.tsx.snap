// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describing MainLogo.tsx component render MainLogo.tsx test component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#656776",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "color": "#43444e",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans_700Bold",
          },
          Object {
            "textShadowOffset": Object {
              "height": 2,
              "width": -2,
            },
            "textShadowRadius": 1,
          },
          Object {},
        ]
      }
    >
      Vehiclegenics
    </Text>
  </View>
</View>
`;
